### DISCUSSION TITLE
Automatically hide SEO field from Collections that don't have URLs?

### DISCUSSION THREAD

**Discussion by aaronbushnell** [aerni/statamic-advanced-seo#136] [Ideas]
One thing that might be nice for Advanced SEO to pick up on is if the Collection has URLs and to hide the SEO tab from those entries if so.

I did see the `advanced-seo.php` config allows for an array of `disabled` collections, but I think most of the ones I'd add in this list would be due to that URL convention.

**Comment by aerni** [comment]
That's a good idea. Adding it to my list of features for the future.

**Comment by aerni** [comment]
I considered auto-disabling SEO for collections without routes, but decided against it for a few reasons:

**Headless setups**
When using a headless frontend, Statamic collections often don't have routes defined, but SEO data is still valuable (meta tags, OpenGraph, etc. get consumed via API).

**Multi-site complexity**
A collection might have routes on Site A but not Site B. Per-site granularity adds significant complexity.

**Edge cases**
Some users have legitimate use cases for SEO on routeless collections (relationship-only collections that surface data elsewhere, future route plans, etc.).

**Simplicity**
Auto-detection logic has edge cases that can frustrate users when it guesses wrong. Manual control is more predictable.

In v3, I'm replacing the disabled config array with a new config-based collection/taxonomy config screen. Each SEO set now has its own configuration page where you can enable/disable it, plus quick enable/disable actions directly in the CP dropdown menu. Much better UX than digging through config files.
